app/mailers/application_mailer.rb

```diff
@@ -1,3 +1,28 @@
 class ApplicationMailer < Mail::Notify::Mailer
+  rescue_from Notifications::Client::BadRequestError, with: :handle_notify_exception
+
   GENERIC_NOTIFY_TEMPLATE = '2744ea53-34f1-431f-8173-8388fadd826a'.freeze
+
+  def handle_notify_exception(e)
+    exception = case e.message
+                when 'BadRequestError: Can’t send to this recipient using a team-only API key'
+                  NotifyTeamOnlyAPIKeyError.new(e.message)
+                else
+                  NotifyOtherBadRequestError.new
```

Comment:
We should pass in `e.message` here too and use that instead of the `DEFAULT_MESSAGE`.